Add Django redirects app to allow URLs to be redirected. Fixes #1946

tompollard opened the pull request:
We often find the external websites, articles, etc, include links to non-existent URLs on the physionet.org domain. This pull request adds the Django Redirects App, allowing us to set up URL redirects (addressing #1946). 

The changes are:

1) Add the Redirects App, following instructions at: https://github.com/MIT-LCP/physionet-build/issues/1946
2) Add a test to confirm that setting up a redirect results in URLs being redirected.
3) Display a new page in the console that lists redirects: http://localhost:8000/console/redirects/. The page is listed under the "Pages" item in the console menu. Not an ideal place, but the best I could think of. The whole menu needs a clean up.
4) Grant the admin user the permission necessary to view the redirects page in the console (`redirects.redirect.view_redirect`).

Couple of notes:

- We will need to add the new permission (`redirects.redirect.view_redirect`) to relevant permissions groups if/once the change is merged.
- For now, redirects will need to be added via the shell using the syntax below:

```
from django.conf import settings
from django.contrib.redirects.models import Redirect
# Add a new redirect.
redirect = Redirect.objects.create(site_id=settings.SITE_ID,
                                   old_path='/contact-us/',
                                   new_path='/contact/')
```

bemoody commented:
The Site stuff is weird.  IIUC, this just ignores the Host (because SITE_ID is set)?  (So following your example it would return the same response for `https://physionet.org/contact-us/`, `https://alpha.physionet.org/contact-us/`, etc.?)

tompollard commented:
I can look into it, but setting the `site_id` seemed necessary.

If you update `site_id` in physionet-django/physionet/test_urls.py to something other than `settings.SITE_ID`, then the redirect will not work. 

e.g. if `site_id` is set to 1 (alpha.physionet.org) in the following chunk, the second `response.status_code` will return a 404:

```
    def test_redirect(self):
        """
        Creates and tests a redirect.
        """
        site_id = settings.SITE_ID
        old_path = '/redirect/me/'
        new_path = '/news/'

        response = self.client.get(old_path)
        self.assertEqual(response.status_code, 404)

        self._add_redirect(site_id=site_id, old_path=old_path, new_path=new_path)

        response = self.client.get(old_path)
        self.assertEqual(response.status_code, 301)
```

tompollard commented:
(@bemoody Let me know if I missed your point...re-reading, I wonder if I did?)

bemoody commented:
Looks good!

In the test case:
```
+    def test_redirect(self):
+        """
+        Creates and tests a redirect.
+        """
+
+        ...
+
+        redirect = Redirect.objects.get(old_path=old_path)
+        redirect.delete()
```
There is no need to delete objects here; Django's TestCase rolls back all database changes after running each test case.  This class also doesn't need to inherit from TestMixin, just TestCase.  (We ought to rename TestMixin...)

tompollard commented:
Thanks @bemoody. I have updated the PR based on your comments above:

> There is no need to delete objects here; Django's TestCase rolls back all database changes after running each test case.

Removed the lines that deleted the objects.

> This class also doesn't need to inherit from TestMixin, just TestCase

Now inheriting from TestCase only.
